### PR TITLE
Move docs on loss function creation to a separate doc file

### PR DESCRIPTION
**Pull request by hroff-1902**
The docs on Hyperopt will be better structured and easier for an ordinal user if we move some sophisticated parts that are rarely needed and require higher coding skills and Python knowledge to a new Advanced Hyperopt chapter. (Only one such part for now actually -- the section describing the Creation of a custom loss function)

### CODE REVIEW

File: docs/advanced-hyperopt.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Advanced Hyperopt
+
+This page explains some advanced Hyperopt issues that may require higher
```
Comment (xmatthias) [inline docs/advanced-hyperopt.md]: > This page explains some advanced Hyperopt issues that ...

i would not call it "issues" - wanting to customize a loss function should not start (nor end) in an issue ...
more like "advanced Hyperopt usecases" - or "advanced Hyperopt applications" (?) ... 
Not necessarily one of the 2 above if you have a better idea...

**Comment by hroff-1902** [comment]
This section need some more love, to be reworked a bit, at this time I specifically simply moved this 1:1 so that other changes will be visible in the changes. I'll apply your notes above 👍 The rest is later in a sep. PR (I need a cleaner Hyperopt page to instert the descriptions for new `hyperopt-list` and `hyperopt-show` commands, that's why this PR goes first)

File: docs/advanced-hyperopt.md (hunk 1 above)
Comment (hroff-1902) [reply docs/advanced-hyperopt.md]: pls review my suggestion for this

**Comment by xmatthias** [comment]
> I need a cleaner Hyperopt page to insert the descriptions for new hyperopt-list and hyperopt-show commands, that's why this PR goes first

Wouldn't you want to document these in utils.md and only link to them from hyperopt?
that's how we're doing it with other utility commands, anyway ...


ps: i changed the "visible part" of the link back ... users will find the file in user_data once create-datadir is called - that's how we did it with other files that moved to templates, too. 
users will not find `freqtrade/templates/` if installed via pypi ... ;)

**Pull request merged**